qiime2/metadata.py

```diff
@@ -14,6 +14,38 @@
 
 class Metadata:
     def __init__(self, dataframe):
+        # `/` and `\0` aren't permitted because they are invalid filename
+        # characters on *nix filesystems. The remaining values aren't permitted
+        # because they *could* be misinterpreted by a shell (e.g. `*`, `|`).
+        illegal_chars = ['/', '\0', '\\', '*', '<', '>', '?', '|', '$']
+        illegal_chars = set(illegal_chars)
+        chars_for_msg = ", ".join("%r" % i for i in illegal_chars)
```

Comment:
Actually it might be better to leave the list intact so that the message isn't stochastic.